tests/test_volume.py

```diff
@@ -120,6 +130,106 @@ def testProject(self):
+        ref_vol = np.zeros((8, L, L, L))
+        index = 0
+        for i in range(2):
+            for j in range(2):
+                for k in range(2):
+                    ref_vol[index, L // 2 + (-1) ** i, L // 2, L // 2] = 1
+                    ref_vol[index, L // 2, L // 2 + (-1) ** j, L // 2] = 1
+                    ref_vol[index, L // 2, L // 2, L // 2 + (-1) ** k] = 1
+                    index += 1
+        ref_vol = Volume(ref_vol)
+
+        # Compare rotated volumes with appropriate reference volumes
+        atol = utest_tolerance(self.dtype)
+        # 3 equivalent cases of rotating by zero degrees about each axis
+        for k in range(3):
+            self.assertTrue(np.allclose(ref_vol[0], rot_vols[4 * k], atol=atol))
+
+        # Equivalent cases of rotation by pi/2 about y-axis and 3*pi/2 about x-axis
+        for k in range(2):
+            self.assertTrue(np.allclose(ref_vol[1], rot_vols[6 * k + 1], atol=atol))
+
+        # Equivalent cases of rotation by pi/2 about x-axis and 3*pi/2 about z-axis
+        for k in range(2):
+            self.assertTrue(np.allclose(ref_vol[2], rot_vols[6 * k + 3], atol=atol))
+
+        # Equivalent cases of rotation by pi/2 about z-axis and 3*pi/2 about y-axis
+        for k in range(2):
+            self.assertTrue(np.allclose(ref_vol[4], rot_vols[6 * k + 5], atol=atol))
+
+        # Rotation by pi about x-axis, pi about y-axis, and pi about z-axis, respectively
+        self.assertTrue(np.allclose(ref_vol[3], rot_vols[2], atol=atol))
+        self.assertTrue(np.allclose(ref_vol[5], rot_vols[6], atol=atol))
+        self.assertTrue(np.allclose(ref_vol[6], rot_vols[10], atol=atol))
+
+    def testCnSymmetricVolume(self):
+        # We create volumes with Cn symmetry and check that they align when rotated by multiples of 2pi/n.
+        L = self.res
+        sym_type = {2: "C2", 3: "C3", 4: "C4", 5: "C5", 6: "C6"}
+
+        for k, s in sym_type.items():
+            # Build rotation matrices that rotate by multiples of 2pi/k about the z axis
+            rot_mat = np.zeros((k, 3, 3), dtype=self.dtype)
+            for i in range(k):
+                rot_mat[i, :, :] = [
+                    [np.cos(2 * i * np.pi / k), -np.sin(2 * i * np.pi / k), 0],
+                    [np.sin(2 * i * np.pi / k), np.cos(2 * i * np.pi / k), 0],
+                    [0, 0, 1],
+                ]
+
+            # To build a reference volume we rotate a volume instance.
+            # ref_vol[0] is a volume rotated by zero degrees.
+            sim = Simulation(L=L, C=1, symmetry_type=s, dtype=np.float64)
+            vol = sim.vols
+            ref_vol = vol.rotate(0, rot_mat, nyquist=False)
+
+            # We rotate ref_vol[0] by the stack of rotation matrices
+            # rot_vol is a stack of rotated ref_vol[0]
+            rot_vol = ref_vol.rotate(0, rot_mat, nyquist=False)
+
+            # Compare rotated volumes to reference volume within the shpere of radius L/4.
```

Comment:
Why?

Reply:
This is because rotated volumes are not accurate in the corners. This is due to our gaussian_blobs not being restricted to the unit sphere. I created issue #521 to address this.

Reply:
Thanks!